### ISSUE TITLE
Elasticsearch init-keystore fails on container restart

### ISSUE THREAD

**Issue by nathan-c** [elastic/cloud-on-k8s#4504]
## Bug Report
**What did you do?**
Added GCP credentials secret as a secureSetting and then restarted cluster

**What did you expect to see?**
Cluster restart successfully

**What did you see instead? Under which circumstances?**
Nodes failed to start successfully needing multiple pod restarts due to a combination of what I believe is a bug in the init keystore script and the readiness probe initialDelaySeconds being too low for my k8s nodes.

The readiness probe regularly causes pod restarts and the init-container script logic to handle this doesn't seem to be working.
The following line of code is intended to be used to stop the init-container re-running the keystore initialisation but the file used is placed in a configMap volume and I believe the flag is being removed on container restart
https://github.com/elastic/cloud-on-k8s/blob/f52ca4700f56238c56186f0fc751ada563cf6dda/pkg/controller/common/keystore/initcontainer.go#L39

It my be worth mentioning that I am changing the default resource allocations for the init-containers as the 500m request causes my 2vCPU nodepool to scale up unnecessarily 

**Environment**

* ECK version:
1.5.0

* Kubernetes information:
GKE: 1.18.17-gke.100
Nodes: 3x e2-custom-2-9216-pool
```
Client Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.18-dispatcher", GitCommit:"de944d802c49735ad0f4bfe82ddfa19737ebe962", GitTreeState:"clean", BuildDate:"2021-05-13T17:51:55Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.17-gke.100", GitCommit:"c702fe983d18291df698149425102b51a8258b71", GitTreeState:"clean", BuildDate:"2021-03-19T09:19:23Z", GoVersion:"go1.13.15b4", Compiler:"gc", Platform:"linux/amd64"}
```

* Resource definition:
```
kind: Elasticsearch
apiVersion: elasticsearch.k8s.elastic.co/v1
metadata:
  annotations:
    common.k8s.elastic.co/controller-version: 1.5.0
    elasticsearch.k8s.elastic.co/cluster-uuid: <REDACTED>
    meta.helm.sh/release-name: <REDACTED>
    meta.helm.sh/release-namespace: <REDACTED>
  creationTimestamp: '2021-03-20T18:24:42Z'
  generation: 9
  labels:
    app.kubernetes.io/managed-by: Helm
  name: elastic
  namespace: <REDACTED>
spec:
  auth: {}
  http:
    service:
      metadata:
        creationTimestamp: null
      spec: {}
    tls:
      certificate: {}
  nodeSets:
    - count: 3
      name: default
      podTemplate:
        spec:
          affinity:
            nodeAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                nodeSelectorTerms:
                  - matchExpressions:
                      - key: cloud.google.com/gke-preemptible
                        operator: DoesNotExist
            podAntiAffinity:
              preferredDuringSchedulingIgnoredDuringExecution:
                - podAffinityTerm:
                    labelSelector:
                      matchLabels:
                        common.k8s.elastic.co/type: elasticsearch
                        elasticsearch.k8s.elastic.co/cluster-name: elastic
                    topologyKey: kubernetes.io/hostname
                  weight: 100
                - podAffinityTerm:
                    labelSelector:
                      matchLabels:
                        common.k8s.elastic.co/type: elasticsearch
                        elasticsearch.k8s.elastic.co/cluster-name: elastic
                    topologyKey: topology.kubernetes.io/zone
                  weight: 80
          containers:
            - env:
                - name: ES_JAVA_OPTS
                  value: '-Xms512m -Xmx512m'
              name: elasticsearch
              resources:
                limits:
                  memory: 1Gi
                requests:
                  cpu: 10m
                  memory: 1Gi
          initContainers:
            - command:
                - sh
                - '-c'
                - sysctl -w vm.max_map_count=262144
              name: sysctl
              securityContext:
                privileged: true
            - command:
                - sh
                - '-c'
                - bin/elasticsearch-plugin install --batch repository-gcs
              name: install-plugins
            - name: elastic-internal-init-filesystem
              resources:
                limits:
                  cpu: 1
                  memory: 50Mi
                requests:
                  cpu: 1m
                  memory: 50Mi
            - name: elastic-internal-init-keystore
              resources:
                limits:
                  cpu: 1
                requests:
                  cpu: 1m
                  memory: 50Mi
      volumeClaimTemplates:
        - metadata:
            name: elasticsearch-data
          spec:
            accessModes:
              - ReadWriteOnce
            resources:
              requests:
                storage: 10Gi
            storageClassName: premium-rwo
  secureSettings:
    - entries:
        - key: key.json
          path: gcs.client.default.credentials_file
      secretName: testusa1-es-backups
  transport:
    service:
      metadata:
        creationTimestamp: null
      spec: {}
    tls:
      certificate: {}
  updateStrategy:
    changeBudget: {}
  version: 7.10.2
status:
  availableNodes: 2
  health: yellow
  phase: ApplyingChanges
  version: 7.10.2

```

* Logs:
```
+ keystore_initialized_flag=/usr/share/elasticsearch/config/elastic-internal-init-keystore.ok
+ [[ -f /usr/share/elasticsearch/config/elastic-internal-init-keystore.ok ]]
+ echo 'Initializing keystore.'
Initializing keystore.
+ /usr/share/elasticsearch/bin/elasticsearch-keystore create
Exception in thread "main" java.lang.IllegalStateException: unable to read from standard input; is standard input open and a tty attached?
at org.elasticsearch.cli.Terminal$SystemTerminal.readText(Terminal.java:273)
at org.elasticsearch.cli.Terminal.promptYesNo(Terminal.java:152)
at org.elasticsearch.common.settings.CreateKeyStoreCommand.execute(CreateKeyStoreCommand.java:51)
at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86)
at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:127)
at org.elasticsearch.cli.MultiCommand.execute(MultiCommand.java:91)
at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:127)
at org.elasticsearch.cli.Command.main(Command.java:90)
at org.elasticsearch.common.settings.KeyStoreCli.main(KeyStoreCli.java:43)
```

Edit: removed memory limit as it was causing container OOM which is unrelated to actual bug

**Comment by nathan-c** [comment]
I have found a workaround is to change the readinessProbe as follows:
```
          containers:
            - name: elasticsearch
              readinessProbe:
                exec:
                  command:
                    - bash
                    - '-c'
                    - /mnt/elastic-internal/scripts/readiness-probe-script.sh
                initialDelaySeconds: 60
                timeoutSeconds: 5
                periodSeconds: 10
                successThreshold: 1
                failureThreshold: 5
```

**Comment by sebgl** [comment]
> The following line of code is intended to be used to stop the init-container re-running the keystore initialisation but the file used is placed in a configMap volume and I believe the flag is being removed on container restart
cloud

I double-checked locally, this file is placed in an emptyDir volume:

```
# bash in the pod
[root@elasticsearch-sample-es-default-0 config]# ls -la /usr/share/elasticsearch/config/
total 64
drwxrwxrwx 7 root          root  4096 May 25 08:57 .
drwxrwxr-x 1 elasticsearch root  4096 May 25 08:57 ..
-rw-r--r-- 1 root          root     0 May 25 08:57 elastic-internal-init-keystore.ok
-rw-rw---- 1 root          root   267 May 25 08:57 elasticsearch.keystore
lrwxrwxrwx 1 root          root    60 May 25 08:56 elasticsearch.yml -> /mnt/elastic-internal/elasticsearch-config/elasticsearch.yml
drwxrwxrwt 3 root          root   140 May 25 08:56 http-certs
-rw-rw-r-- 1 elasticsearch root  3110 Feb  8 22:41 jvm.options
drwxrwxr-x 2 elasticsearch root  4096 Feb  8 22:45 jvm.options.d
-rw-rw-r-- 1 elasticsearch root 18535 Feb  8 22:46 log4j2.file.properties
-rw-rw-r-- 1 elasticsearch root  9970 Feb  8 22:50 log4j2.properties
drwxr-xr-x 2 root          root  4096 May 25 08:56 node-transport-cert
-rw-rw-r-- 1 elasticsearch root   473 Feb  8 22:46 role_mapping.yml
lrwxrwxrwx 1 root          root    48 May 25 08:56 roles.yml -> /mnt/elastic-internal/xpack-file-realm/roles.yml
drwxrwxrwt 3 root          root   140 May 25 08:56 transport-certs
drwxrwxrwt 3 root          root   100 May 25 08:56 transport-remote-certs
lrwxrwxrwx 1 root          root    53 May 25 08:56 unicast_hosts.txt -> /mnt/elastic-internal/unicast-hosts/unicast_hosts.txt
lrwxrwxrwx 1 root          root    44 May 25 08:56 users -> /mnt/elastic-internal/xpack-file-realm/users
lrwxrwxrwx 1 root          root    50 May 25 08:56 users_roles -> /mnt/elastic-internal/xpack-file-realm/users_roles
```

```
# volumes
 {
      "emptyDir": {},
      "name": "elastic-internal-elasticsearch-config-local"
    },
```

```
# volume mounts
 {
          "mountPath": "/usr/share/elasticsearch/config",
          "name": "elastic-internal-elasticsearch-config-local"
        },
```

**Comment by nathan-c** [comment]
@sebgl It looks like you are right. Perhaps I didn't realise my memory limit was being hit so the keystore init was never finishing properly. I can't to reproduce without the memory limit. I thought I saw the problem before i changed the limits from their defaults but I have just tried again using default limits and the cluster comes up okay (a few readinessProbe failures but then okay). Sorry for the confusion!
